packages/core/src/internal/data-grid/data-grid.tsx

```diff
@@ -515,7 +515,7 @@ const DataGrid: React.ForwardRefRenderFunction<DataGridRef, DataGridProps> = (p,
             const isTouch = (ev !== undefined && !(ev instanceof MouseEvent)) || (ev as any)?.pointerType === "touch";
 
             const scrollEdge: GridMouseEventArgs["scrollEdge"] = [
-                x < stickyX ? -1 : rect.width < x ? 1 : 0,
+                x < 0 ? -1 : rect.width < x ? 1 : 0,
```

Comment:
Ideally this would know if the current selection is inside the freeze region or not, but since this is not currently known in this region of the code I didn't want to plumb it through. We could then conditionally scroll. For now however this will have to do.

Reply:
The problematic scenario here is that with `x < 0` you don't get autoscroll until you rmouse leaves the grid entirely. Ideally auto-scroll would start if you go over a sticky column and the current selected cell is not in the sticky region.